hpx/lcos/latch.hpp

```diff
@@ -0,0 +1,143 @@
+//  Copyright (c) 2007-2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_LCOS_LATCH_APR_19_2015_1002AM)
+#define HPX_LCOS_LATCH_APR_19_2015_1002AM
+
+#include <hpx/exception.hpp>
+#include <hpx/include/client.hpp>
+#include <hpx/lcos/server/latch.hpp>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace lcos
+{
+    class latch : public components::client_base<latch, lcos::server::latch>
+    {
+        typedef components::client_base<latch, lcos::server::latch> base_type;
+
+    public:
+        latch()
+        {}
+
+        /// Initialize the latch
+        ///
+        /// Requires: count >= 0.
+        /// Synchronization: None
+        /// Postconditions: counter_ == count.
+        ///
+        explicit latch(std::ptrdiff_t count)
```

Comment:
As I said, it can go either way as far as I care, I'm just presenting you with the argument that is used to support this design. As for `vector`'s index being unsigned, that has been recognized as a design error. Never heard anything about its size argument, but when it comes to arrays the precondition is checked on the argument expression, not the already converted to unsigned parameter.